src/librustdoc/html/static/main.js

```diff
@@ -715,6 +715,13 @@
         }
 
         function startSearch() {
+
+            if ($('.search-input').val().length === 0) {
```

Comment:
This now is not inside an event handler anymore.

Could you change this into 

```javascript
$(".search-input").on("input", function() {
    if ($('.search-input').val().length === 0) {
        window.history.pushState("", "std - Rust", "?search=");
        $('#main.content').removeClass('hidden');
        $('#search.content').addClass('hidden');
    }
});
```

---

After this and https://github.com/rust-lang/rust/pull/28725/files#r40717663 are resolved, this PR LGTM.